config/routes.rb

```diff
@@ -1,13 +1,12 @@
 Rails.application.routes.draw do
-  root "users#index"
-  get "/users/", to: "users#index"
-  get "/users/:id", to: "users#show"
-  get "/users/:id/posts", to: "users#list"
-  get "/users/:id/posts/:id", to: "users#posts"
+  resources :users, only: %i[show index] do
+    resources :posts, only: %i[show create new index]
+  end
 
+  root to: 'users#index'
 
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+  resources :posts do
+    resources :comments, only: %i[create new]
+    resources :likes, only: %i[create]
+  end
```

Comment:
- [ ] Currently, you have two separate resources for the `Post` controller. I would like to advise you to have only one resource. You can decide to adopt the code below:

```
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create, :destroy] do
      resources :comments, shallow: true
      resources :likes, only: [:new, :create]
    end
  end
```